Fixed Compiler errors, updated to newest java version, updated travis, updated to newest featureIDE

wurstbroteater opened the pull request:
fixed #45 
fixed #46 
fixed #47
fixed #48 
fixed  #42 
During the review, it became apparent that there problems concerning the code style and documentation. This caused the new issues #56 and #57

wurstbroteater committed:
Deleted setPossibleWords(...)
This method has been deleted in FeatureIDE since this commit
https://github.com/FeatureIDE/FeatureIDE/commit/b679c392ba6bd29afb47d1b8071a1b337d90e697#diff-02b7a28d524fe3c717b03a9621030e760993f27c394403d3d837e36e4c4489fa
Since there is no provision of a new method, it can be deleted.
fixed #48

tcerny committed:
Merge pull request #1 from wurstbroteater/issue-48

Deleted setPossibleWords(...)

tcerny committed:
fixed #46

jeremiaheinle committed:
bugfix issue-47

tcerny committed:
Merge remote-tracking branch 'origin/issue47' into issue-45

tcerny committed:
Merge remote-tracking branch 'origin/issue-48' into issue-45

tcerny committed:
added missing method initExtension()

jeremiaheinle committed:
DeltaFactoryManager update

wurstbroteater committed:
updated travis and pom

wurstbroteater committed:
forgotten commit in issue-45

wurstbroteater committed:
Tests do not trigger has been fixed

wurstbroteater committed:
this branch fixed #42

jeremiaheinle committed:
Issue initExtension & DeltaFactoryManager

wurstbroteater committed:
updated travis services to run with bionic and increased tycho version in pom

wurstbroteater committed:
deleted unnecessary depdency

tcerny committed:
Casted Path to IFile in order to keep the old code.

tcerny committed:
Merge branch 'issue-46' into issue-45

tcerny committed:
Merge branch 'issue-45' of github.com:tcerny/VariantSync into issue-45

wurstbroteater committed:
Update .travis.yml

wurstbroteater committed:
win32 x86 no longer supported

wurstbroteater committed:
works on my machine with tests and java >= 11

tcerny committed:
Merge branch 'issue-42' into issue-45

tcerny committed:
Update .gitignore

jeremiaheinle committed:
del .idea

tcerny committed:
Update .gitignore

wurstbroteater committed:
Merge pull request #2 from tcerny/issue-45

accpted YaY

pmbittner commented:
Hi Eric, do you see that?

pmbittner commented:
Hi Tobi, do you see that?

pmbittner commented on de.tubs.variantsync.core-test/test/de/tubs/variantsync/patch/base/DiffUtilsTest.java:
```diff
@@ -54,6 +54,7 @@ public void setUp() throws Exception {
 	public void testPatch1() throws DiffException {
 		Patch<String> patch = DiffUtils.diff(lines1, lines3, 1);
 		assertTrue(patch.getDeltas().size() == 3);
+		
```
Why did you add the empty line here? Please fix.

wurstbroteater replied:
fixed: https://github.com/tthuem/VariantSync/pull/53/commits/02b4399e94925cbea7ba19cd021e9e846522bc85

pmbittner commented on .travis.yml:
```diff
@@ -1,14 +1,18 @@
+language: java 
+jdk:
+  - oraclejdk11
+  - openjdk10
```
As discussed: Please remove openjdk10 because it fails (and seems deprecated here).

wurstbroteater replied:
fixed https://github.com/tthuem/VariantSync/pull/53/commits/cbfe2a3cf2b3b665254c6ab6d6b802179d5c2fde

wurstbroteater committed:
adressed: Pauls review: remove openjdk10

wurstbroteater committed:
adressed Pauls review: remove empy line

pmbittner reviewed (approved):
Thanks for the fixes.

h3ssto commented on .travis.yml:
```diff
@@ -1,14 +1,17 @@
+
+language: java 
+jdk:
+  - oraclejdk11
```
remove `oraclejdk11` add `openjdk8`

wurstbroteater replied:
(Open) JDK 8 not possible because newest Tycho/Maven/some-Plugin-there version does not support 8 anymore
```
[WARNING] Error injecting: org.eclipse.tycho.core.p2.P2ArtifactRepositoryLayout
java.lang.TypeNotPresentException: Type org.eclipse.tycho.core.p2.P2ArtifactRepositoryLayout not present
```
....
```
 java.lang.UnsupportedClassVersionError: org/eclipse/tycho/core/p2/P2ArtifactRepositoryLayout has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
```

travis failed build: https://travis-ci.org/github/tcerny/VariantSync/jobs/751325693

EDIT: 
it fails with JDK10 aswell https://travis-ci.org/github/tcerny/VariantSync/builds/753209334
ATM only working JDK is >= 11

h3ssto replied:
Then keep it at jdk11-openjdk.

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/VariantSyncComposer.java:
```diff
@@ -15,7 +18,8 @@
 	@Override
-	public void performFullBuild(IFile config) {
+	public void performFullBuild(Path config) {
+//		public void performFullBuild(IFile config) {
```
Remove quoted line

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/VariantSyncComposer.java:
```diff
@@ -90,4 +94,17 @@ public void copyNotComposedFiles(Configuration config, IFolder destination) {
+	@Override
+	public void buildPartialFeatureProjectAssets(IFolder sourceFolder, ArrayList<String> removedFeatures,
+			ArrayList<String> mandatoryFeatures) throws IOException, CoreException {
+		// TODO Auto-generated method stub
```
remove TODO

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/VariantSyncComposer.java:
```diff
@@ -90,4 +94,17 @@ public void copyNotComposedFiles(Configuration config, IFolder destination) {
+
+	@Override
+	public boolean supportsPartialFeatureProject() {
+		// TODO Auto-generated method stub
```
Remove TODO, explain why this method always returns `false`. Instead of this pattern, I would prefer moving this method into an interface and resolving this problem on compile time.

tcerny replied:
I don't exactly understand what you mean with moving the method into an interface and resolving this problem on compile time.
The method is required by the `ComposerExtensionClass` which is extended by this class (respectively the `IComposerExtensionBase` interface).
The method always returns `false` because no partial Feature Project is needed within VariantSync.

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/managers/ConfigurationProjectManager.java:
```diff
@@ -115,7 +119,8 @@ public ConfigurationProject getConfigurationProject(IFeatureProject project) {
 	private void findVariants(ConfigurationProject configurationProject) {
-		for (IFile file : configurationProject.getFeatureProject().getAllConfigurations()) {
+		for (Path path : configurationProject.getFeatureProject().getAllConfigurations()) {
+			IFile file = (IFile) EclipseFileSystem.getResource(path);
```
Is this type cast always possible?

tcerny replied:
In my opinion this type cast should be always possible because the `getAllConfigurations()` method should check wether the file is supported, i.e. if it is a valid configuration file. And config files can be treated as `IFile`.
Furthermore this type cast is used even in FeatureIDE, e.g. in `plugins/de.ovgu.featureide.core.ahead/src/de/ovgu/featureide/ahead/AheadComposer.java`

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/managers/data/ConfigurationProject.java:
```diff
@@ -75,13 +76,14 @@ public void addVariant(IProject project) {
 		if (project != null) {
-			for (IFile configPath : configurationProject.getAllConfigurations()) {
+			for (Path confPath : configurationProject.getAllConfigurations()) {
+				IFile configPath = (IFile) EclipseFileSystem.getResource(confPath);
```
Is this type cast always possible?

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/managers/data/ConfigurationProject.java:
```diff
@@ -75,13 +76,14 @@ public void addVariant(IProject project) {
 				if (configName.equals(project.getName())) {
 					ConfigurationManager configurationManager = ConfigurationManager.getInstance(Paths.get(configPath.getRawLocationURI()));
-					if (configurationManager != null) return configurationManager.getObject();
+					if (configurationManager != null) return configurationManager.getObject();			
```
This is against basic style guides, all optional brackets should be written. @pmbittner I thought you copied the FeatureIDE style config to VariantSync?

pmbittner replied:
I did but I don't know how if these files are considered by IntelliJ and I don't know if you have to import them manually in Eclipse and I don't think the style files will make the IDE rewriting existing code on its own.

tcerny replied:
We had to import the code style xml file (Settings -> Java -> CodeStyle -> Formatter -> Import) first. 
We will fix this in future and rerun the code formatter on all other files.

wurstbroteater replied:
In Intellij File -> Settings -> Editor -> Code Style -> Java -> import eclipse scheme

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/managers/persistence/CodeMappingFormat.java:
```diff
@@ -35,7 +36,7 @@ public CodeMappingFormat(IProject project) {
 
 	@Override
-	public IPersistentFormat<List<SourceFile>> getInstance() {
+	public APersistentFormat<List<SourceFile>> getInstance() {
```
I do not understand this change.

jeremiaheinle replied:
Because of changes in featureide,
IPersistentFormat.getInstance gets further specialized into APersistentFormat.getInstance
and cant be regeneralized later on.
AXMLFormat has unecessary double inheritance, it inherits from IPersistentFormat and extends APersistentFormat, so it is unclear what to implement.

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/managers/persistence/FeatureContextFormat.java:
```diff
@@ -23,7 +23,7 @@
 
 	@Override
-	public IPersistentFormat<List<FeatureContext>> getInstance() {
+	public APersistentFormat<List<FeatureContext>> getInstance() {
```
I do not understand this change.

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/managers/persistence/PatchFormat.java:
```diff
@@ -44,7 +45,7 @@ public PatchFormat(IFeatureProject project) {
 	
 	@Override
-	public IPersistentFormat<List<IPatch<?>>> getInstance() {
+	public APersistentFormat<List<IPatch<?>>> getInstance() {
```
I do not understand this change.

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/patch/DeltaFactoryManager.java:
```diff
@@ -13,9 +20,12 @@
 	public DeltaFactoryManager() {
 		IDeltaFactory[] y = new IDeltaFactory[]{getDefaultFactory()};
-		IExtensionLoader<IDeltaFactory> x = new CoreExtensionLoader<IDeltaFactory>(y);
+		IExtensionLoader<IDeltaFactory> x = new EclipseExtensionLoader<IDeltaFactory>(VariantSyncPlugin.PLUGIN_ID, IDeltaFactory.extensionPointID, IDeltaFactory.extensionID, IDeltaFactory.class);
```
Please add line breaks to make this constructor more readable.

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/patch/DeltaFactoryManager.java:
```diff
@@ -63,5 +73,53 @@ public IDeltaFactory getFactoryByFile(IFile file) throws NoSuchExtensionExceptio
+			}
+		}
+		return Collections.unmodifiableList(extensions);
+		}
```
The indents look all over the place, tab size = 4 and proper indention should be enforced by the style config @pmbittner

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/patch/DeltaFactoryManager.java:
```diff
@@ -63,5 +73,53 @@ public IDeltaFactory getFactoryByFile(IFile file) throws NoSuchExtensionExceptio
+		}
+	}
+	return false;
+}
```
Same here, please fix the indentation by using the style config from FeatureIDE.

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/patch/DeltaFactoryManager.java:
```diff
@@ -63,5 +73,53 @@ public IDeltaFactory getFactoryByFile(IFile file) throws NoSuchExtensionExceptio
+		}
+	}
+	throw new NoSuchExtensionException("No extension found for ID " + id);
+}
```
Style.

h3ssto reviewed (changes requested):
The CI stuff (Travis, Maven) looks really neat and is approved. Parts of the Java code lack explanation and style. See inline comments.

tcerny committed:
update .gitignore

tcerny committed:
addressed Tobi's review
- removed quoted lines in VariantSyncComposer.java
- added optional brackets in ConfigurationProject.java

wurstbroteater committed:
adressed Tobis review

added openjdk8 to reproduce error. Removed ocralejdk

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/VariantSyncComposer.java:
```diff
@@ -17,93 +17,73 @@
 
-	@Override
-	public boolean hasSourceFolder() {
+	@Override public boolean hasSourceFolder() {
```
Why is the `@Override` inline?

wurstbroteater replied:
This is  the output if the provided code style is used (with automated code  formatting)

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/VariantSyncComposer.java:
```diff
@@ -2,92 +2,89 @@
+	}
+
+	@Override public boolean supportsPartialFeatureProject() {
+		return false;
```
Indent all over the place (tabsize = 8?)

h3ssto replied:
?

Edit by @pmbittner : Responding to this comment is impossible in Github. The same comment was made earlier here (https://github.com/tthuem/VariantSync/pull/53#discussion_r557359440).

tcerny replied:
this has already been fixed, but Github doesn't track the changes for some reason

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/patch/DeltaFactoryManager.java:
```diff
@@ -1,10 +1,17 @@
 import org.eclipse.core.resources.IFile;
 
-import de.ovgu.featureide.fm.core.CoreExtensionLoader;
+//import de.ovgu.featureide.fm.core.CoreExtensionLoader;
```
Remove commented import.

h3ssto commented:
@wurstbroteater @tcerny @jeremiaheinle Many of the requested changes have not been addressed, e.g. code style is still in an abysmal state, random lines are commented out, without any additional annotation.

pmbittner commented on .travis.yml:
```diff
@@ -1,14 +1,17 @@
+#-----start GUI testing stuff
+services:
+  - xvfb
+#-----end GUI testin stuff
```
fix typo and remove "stuff" please

wurstbroteater replied:
This has already been fixed. See: https://github.com/tcerny/VariantSync/tree/issue-52

wurstbroteater committed:
Adressing reviewer suggetions now in correct branch for #53

pmbittner commented on .travis.yml:
```diff
@@ -2,13 +2,12 @@ dist: bionic
 services:
   - xvfb
-#-----end GUI testin stuff
+#-----end GUI testing stuff
```
Also remove "stuff" in both comments, please.

wurstbroteater committed:
Adressing reviewer suggetions for travis

wurstbroteater committed:
Adressing reviewer suggestion to remove commented import

wurstbroteater committed:
Fixed Code Style

wurstbroteater committed:
added build status to readme

pmbittner commented on de.tubs.variantsync.core/.settings/org.eclipse.jdt.core.prefs:
```diff
@@ -1,4 +1,14 @@
 eclipse.preferences.version=1
```
@h3ssto Is it usual for eclipse projects to submit the .settings directory to vcs? Should it be removed?

h3ssto replied:
No it is not common and yes it should not be part of the repo.

pmbittner replied:
Ok, so please remove this folder and add it to the gitignores please @tcerny @jeremiaheinle @wurstbroteater .

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/VariantSyncPlugin.java:
```diff
@@ -60,18 +60,21 @@
-		DeltaFactoryManager
-				.setExtensionLoader(new EclipseExtensionLoader<>(PLUGIN_ID, IDeltaFactory.extensionPointID, IDeltaFactory.extensionID, IDeltaFactory.class));
+		DeltaFactoryManager.setExtensionLoader(new EclipseExtensionLoader<>(PLUGIN_ID, IDeltaFactory.extensionPointID,
+				IDeltaFactory.extensionID, IDeltaFactory.class));
```
Sieht wild aus.

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/VariantSyncPlugin.java:
```diff
@@ -160,9 +167,11 @@ public static ITextEditor getEditor() {
 	public static IFile getEditorInput() {
-		if (VariantSyncPlugin.getActiveWorkbenchWindow() == null) return null;
+		if (VariantSyncPlugin.getActiveWorkbenchWindow() == null)
+			return null;
```
Brackets missing.

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/VariantSyncPlugin.java:
```diff
@@ -160,9 +167,11 @@ public static ITextEditor getEditor() {
 		IEditorPart editorPart = VariantSyncPlugin.getActiveWorkbenchWindow().getActivePage().getActiveEditor();
-		if (editorPart == null || !(editorPart instanceof IFileEditorInput)) return null;
+		if (editorPart == null || !(editorPart instanceof IFileEditorInput))
+			return null;
```
Brackets missing.

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/exceptions/ProjectNotFoundException.java:
```diff
@@ -5,25 +5,28 @@
 	@Override
 	public String toString() {
-		return "A " + (type==Type.CONFIGURATION?"configuration project":"variant") + " does not exist in the workspace. " + getMessage();
+		return "A " + (type == Type.CONFIGURATION ? "configuration project" : "variant")
+				+ " does not exist in the workspace. " + getMessage();
```
OK for now, please use `StringBuilder` in the future.

wurstbroteater replied:
This problem will be addressed by the following issue https://github.com/tthuem/VariantSync/issues/56

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/managers/ConfigurationProjectManager.java:
```diff
@@ -137,8 +147,10 @@ private void findVariants(ConfigurationProject configurationProject) {
+			if (project.getName().equals(featureProject.getProjectName()))
+				return featureProject;
+			if (featureProject.getProject().exists() && INSTANCES.get(featureProject).getVariants().contains(project))
+				return featureProject;
```
Brackets missing.

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/managers/FeatureContextManager.java:
```diff
@@ -88,7 +88,8 @@ public void importFeaturesFromModel() throws ProjectNotFoundException {
 			}
-		} else throw new ProjectNotFoundException(Type.CONFIGURATION);
+		} else
+			throw new ProjectNotFoundException(Type.CONFIGURATION);
```
Brackets missing.

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/managers/PatchesManager.java:
```diff
@@ -27,7 +27,8 @@ public PatchesManager(ConfigurationProject configurationProject) {
 	public IPatch<?> getActualContextPatch() {
-		if (actualPatch == null) return null;
+		if (actualPatch == null)
+			return null;
```
Brackets missing.

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/managers/PatchesManager.java:
```diff
@@ -50,7 +51,8 @@ public void setPatches(List<IPatch<?>> patches) {
 			actualPatch.setEndTime(System.currentTimeMillis());
-			if (!actualPatch.isEmpty() && !patches.contains(actualPatch)) patches.add(actualPatch);
+			if (!actualPatch.isEmpty() && !patches.contains(actualPatch))
+				patches.add(actualPatch);
```
Brackets missing.

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/managers/data/ConfigurationProject.java:
```diff
@@ -38,7 +39,8 @@
 	public IFeatureProject getFeatureProject() {
-		return configurationProject != null ? configurationProject.getProject().exists() ? configurationProject : null : null;
+		return configurationProject != null ? configurationProject.getProject().exists() ? configurationProject : null
+				: null;
```
This is error prone, please use if-else-else if.

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/managers/data/ConfigurationProject.java:
```diff
@@ -60,7 +62,8 @@ public void setFeatureProject(IFeatureProject configurationProject) {
 		for (IProject project : this.projects) {
-			if (project.getName().equals(name)) return project;
+			if (project.getName().equals(name))
+				return project;
```
Brackets missing.

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/managers/persistence/CodeMappingFormat.java:
```diff
@@ -24,7 +25,8 @@
 	private static final String CODEMAPPINGS = "CodeMapping";
-	private static final Pattern CONTENT_REGEX = Pattern.compile("\\A\\s*(<[?]xml\\s.*[?]>\\s*)?<" + MAPPINGS + "[\\s>]");
+	private static final Pattern CONTENT_REGEX = Pattern
+			.compile("\\A\\s*(<[?]xml\\s.*[?]>\\s*)?<" + MAPPINGS + "[\\s>]");
```
Comment what this does.

wurstbroteater replied:
This problem will be addressed by the following issue https://github.com/tthuem/VariantSync/issues/56

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/managers/persistence/FeatureContextFormat.java:
```diff
@@ -18,12 +18,13 @@
 	private static final String FEATURE_CONTEXT = "context";
-	private static final Pattern CONTENT_REGEX = Pattern.compile("\\A\\s*(<[?]xml\\s.*[?]>\\s*)?<" + FEATURE_CONTEXTS + "[\\s>]");
+	private static final Pattern CONTENT_REGEX = Pattern
+			.compile("\\A\\s*(<[?]xml\\s.*[?]>\\s*)?<" + FEATURE_CONTEXTS + "[\\s>]");
```
Comment.

wurstbroteater replied:
This problem will be addressed by the following issue https://github.com/tthuem/VariantSync/issues/56

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/managers/persistence/PatchFormat.java:
```diff
@@ -32,7 +33,8 @@
 	private static final String DELTA = "Delta";
-	private static final Pattern CONTENT_REGEX = Pattern.compile("\\A\\s*(<[?]xml\\s.*[?]>\\s*)?<" + PATCHES + "[\\s>]");
+	private static final Pattern CONTENT_REGEX = Pattern
+			.compile("\\A\\s*(<[?]xml\\s.*[?]>\\s*)?<" + PATCHES + "[\\s>]");
```
Comment.

wurstbroteater replied:
This problem will be addressed by the following issue https://github.com/tthuem/VariantSync/issues/56

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/monitor/CodeMappingHandler.java:
```diff
@@ -32,10 +32,12 @@ public static void addCodeMappingsForDeltas(List<IDelta<?>> deltas) {
 				IDeltaFactory<?> deltaFactory = DeltaFactoryManager.getFactoryById(delta.getFactoryId());
 				IMarkerHandler markerHandler = deltaFactory.getMarkerHandler();
-				List<IVariantSyncMarker> variantSyncMarkers = markerHandler.getMarkersForDelta(delta.getResource(), delta);
+				List<IVariantSyncMarker> variantSyncMarkers = markerHandler.getMarkersForDelta(delta.getResource(),
+						delta);
```
Warum nicht nur `delta` übergeben?

tcerny replied:
This problem will be addressed by the following issue #57

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/nature/Variant.java:
```diff
@@ -9,32 +9,32 @@
+	public void configure() throws CoreException {
+		// only called once the nature has been set
+
+		// TODO: Create .variantHistory file
```
Why is this TODO not addressed?

h3ssto replied:
Please open an issue and document this gap.

wurstbroteater replied:
This problem will be addressed by the following issue https://github.com/tthuem/VariantSync/issues/57

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/nature/Variant.java:
```diff
@@ -9,32 +9,32 @@
+	}
+
+	@Override
+	public void deconfigure() throws CoreException {
+		// only called once the nature has been set
+
+		// This should !not! delete the .variantHistory file
+
+		// reset the project configuration...
+	}
```
Why `throws CoreException`, why is this method overridden?

wurstbroteater replied:
This problem will be addressed by the following issue https://github.com/tthuem/VariantSync/issues/57

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/patch/ADelta.java:
```diff
@@ -209,45 +211,67 @@ public String getFactoryId() {
-		if (this == obj) return true;
-		if (obj == null) return false;
-		if (getClass() != obj.getClass()) return false;
+		if (this == obj)
+			return true;
+		if (obj == null)
+			return false;
+		if (getClass() != obj.getClass())
+			return false;
```
Brackets.

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/patch/AMarkerHandler.java:
```diff
@@ -24,6 +24,7 @@
 	public abstract List<IVariantSyncMarker> getMarkers(IFile file, int offset, int length);
 
 	@Override
-	public abstract boolean updateMarkerForDelta(SourceFile sourceFile, IDelta<T> delta, List<IVariantSyncMarker> variantSyncMarkers);
+	public abstract boolean updateMarkerForDelta(SourceFile sourceFile, IDelta<T> delta,
+			List<IVariantSyncMarker> variantSyncMarkers);
 
 }
```
I do not get this `@Override` what is changed?

wurstbroteater replied:
This problem will be addressed by the following issue https://github.com/tthuem/VariantSync/issues/57

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/patch/APatch.java:
```diff
@@ -16,16 +16,20 @@
 	public void addDelta(IDelta<T> delta) {
-		if (delta.getPatch() == null) delta.setPatch(this);
-		if (delta.getContext() == "") delta.setContext(feature);
+		if (delta.getPatch() == null)
+			delta.setPatch(this);
+		if (delta.getContext() == "")
+			delta.setContext(feature);
```
Brackets.

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/patch/APatch.java:
```diff
@@ -16,16 +16,20 @@
 		for (IDelta<T> delta : deltas) {
-			if (delta.getPatch() == null) delta.setPatch(this);
-			if (delta.getContext() == "") delta.setContext(feature);
+			if (delta.getPatch() == null)
+				delta.setPatch(this);
+			if (delta.getContext() == "")
+				delta.setContext(feature);
```
Brackets.

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/patch/HistoryStore.java:
```diff
@@ -15,7 +15,8 @@
 		IProject project = file.getProject();
-		IFile historyFile = project.getFolder(historyFolder).getFolder(file.getProjectRelativePath().toOSString().replace(".", "_").replace("/", "_"))
+		IFile historyFile = project.getFolder(historyFolder)
+				.getFolder(file.getProjectRelativePath().toOSString().replace(".", "_").replace("/", "_"))
```
Document purpose of this in comment.

tcerny replied:
This problem will be addressed by the following issue #57

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/patch/base/DefaultDelta.java:
```diff
@@ -101,8 +101,8 @@ public void setRevisedFromString(String revised) {
 	public String getRepresentation() {
-		return "--- (" + this.original.getPosition() + ") " + this.original.getLines() + "\n" + "+++ (" + this.revised.getPosition() + ") "
-			+ this.revised.getLines();
+		return "--- (" + this.original.getPosition() + ") " + this.original.getLines() + "\n" + "+++ ("
+				+ this.revised.getPosition() + ") " + this.revised.getLines();
```
This is not OS-safe, change `\n` to `String.format("%n");`

pmbittner replied:
Or maybe `System.lineSeparator()`?

wurstbroteater replied:
This problem will be addressed by the following issue https://github.com/tthuem/VariantSync/issues/56

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/patch/base/DefaultDeltaFactory.java:
```diff
@@ -62,9 +62,11 @@ public String toString() {
+			throws DiffException {
 		// Check for null arguments
-		if (res == null || oldState == null || kind == null) return null;
+		if (res == null || oldState == null || kind == null)
+			return null;
```
Brackets.

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/patch/base/DefaultMarkerHandler.java:
```diff
@@ -25,7 +25,8 @@
 		for (IDelta<Chunk<String>> delta : deltas) {
 			Chunk revised = delta.getRevised();
-			IVariantSyncMarker variantSyncMarker = new AMarkerInformation(revised.getPosition() - 1, revised.getLines().size() - 1, true);
+			IVariantSyncMarker variantSyncMarker = new AMarkerInformation(revised.getPosition() - 1,
+					revised.getLines().size() - 1, true);
```
Explain -1 in comment.

wurstbroteater replied:
This problem will be addressed by the following issue https://github.com/tthuem/VariantSync/issues/56

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/syncronization/TargetsCalculator.java:
```diff
@@ -34,8 +34,10 @@
 			Configuration config = configurationProject.getConfigurationForVariant(project);
-			if (config == null || !config.getSelectedFeatureNames().contains(delta.getContext())) continue;
-			if (project != delta.getProject() && isTargetWithoutConflict(project, delta) && !delta.getSynchronizedProjects().contains(project))
+			if (config == null || !config.getSelectedFeatureNames().contains(delta.getContext()))
+				continue;
+			if (project != delta.getProject() && isTargetWithoutConflict(project, delta)
+					&& !delta.getSynchronizedProjects().contains(project))
 				targets.add(project);
```
Brackets.

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/syncronization/TargetsCalculator.java:
```diff
@@ -52,15 +54,18 @@
 			Configuration config = configurationProject.getConfigurationForVariant(project);
-			if (config == null || !config.getSelectedFeatureNames().contains(delta.getContext())) continue;
-			if (project != delta.getProject() && isTargetWithConflict(project, delta) && !delta.getSynchronizedProjects().contains(project))
+			if (config == null || !config.getSelectedFeatureNames().contains(delta.getContext()))
+				continue;
+			if (project != delta.getProject() && isTargetWithConflict(project, delta)
+					&& !delta.getSynchronizedProjects().contains(project))
```
Brackets.

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/utilities/FileHelper.java:
```diff
@@ -87,17 +88,19 @@ public static void setFileLines(IFile res, List<String> lines) {
+	 * List<String>.
 	 *
-	 * @param in buffered Reader for file
+	 * @param in      buffered Reader for file
```
?

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/utilities/FileHelper.java:
```diff
@@ -87,17 +88,19 @@ public static void setFileLines(IFile res, List<String> lines) {
 		String line = "";
 		BufferedReader reader = null;
-		if (charset == null) charset = (String) "UTF-8";
+		if (charset == null)
+			charset = (String) "UTF-8";
```
Brackets.

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/utilities/Tree.java:
```diff
@@ -69,7 +69,8 @@ private TreeNode auxiliaryFind(TreeNode currentNode, Object dataToFind) {
 		int i = 0;
 
-		if (currentNode == null || dataToFind == null) return null;
+		if (currentNode == null || dataToFind == null)
+			return null;
```
Brackets.

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/view/context/ActiveContextHandler.java:
```diff
@@ -29,8 +29,10 @@ public boolean isEnabled() {
 		VariantSyncPlugin.getActiveMappingManager().setActive(!oldValue);
-		if (!oldValue) updateElement.setIcon(VariantSyncPlugin.getDefault().getImageDescriptor("icons/nav_stop.gif"));
-		else updateElement.setIcon(VariantSyncPlugin.getDefault().getImageDescriptor("icons/nav_go.gif"));
+		if (!oldValue)
+			updateElement.setIcon(VariantSyncPlugin.getDefault().getImageDescriptor("icons/nav_stop.gif"));
+		else
+			updateElement.setIcon(VariantSyncPlugin.getDefault().getImageDescriptor("icons/nav_go.gif"));
```
Brackets.

h3ssto commented on de.tubs.variantsync.core/src/de/tubs/variantsync/core/view/featurecontext/FeatureContextHandler.java:
```diff
@@ -17,10 +17,12 @@
 
 	@Override
-	public void addHandlerListener(IHandlerListener handlerListener) {}
+	public void addHandlerListener(IHandlerListener handlerListener) {
+	}
```
I guess this is stubbing for futur work, please document in dedicated issue.

wurstbroteater replied:
This problem will be addressed by the following issue https://github.com/tthuem/VariantSync/issues/57

h3ssto reviewed (changes requested):
Short: I don't see any progress.

Code Style: (reference: https://google.github.io/styleguide/javaguide.html)
* Indents are still quite often wild (please fix to tabsize = 4 spaces)
* All optional brackets must be given to avoid ambiguity ("dangling else")
* There is a plethora of undocumented magic strings and numbers, their purpose should be documented in form of a comment

-> If the style config is broken, then **fix** it.

General:
* Many methods are overridden but empty. Please document this stubbing in issues, so that it can be addressed in the future. 
* Comment! Comment! Comment! Everything that is not self-explanatory should be commented, especially if it looks erroneous. There are over 40 stubbed methods, whose purpose is unclear.

tcerny committed:
Delete de.tubs.variantsync.core-test/.settings directory

tcerny committed:
Delete de.tubs.variantsync.core/.settings directory

tcerny committed:
fixed code style and added all optional brackets

h3ssto commented:
Please ping me, when this PR is ready to be reviewed again. Please also add the newly generated issues to the PR commit message.

wurstbroteater committed:
changed String to StringBuilder

wurstbroteater committed:
Auto format VariantSyncComposer

wurstbroteater committed:
addressing reviewer suggestion: replaced ternary operator with ifElse and Code Style for VSComposer

h3ssto reviewed (approved):
Finally, the golden sunshine of cleanly styled code has touched this battered code base. It is my delight, as his majesties reviewer, to approve this pull request and welcome it into its place in commit history.

*Record screech*

A few comments nevertheless:
- There are still commented out methods, else-if-cases and other stuff in the code base, please clean them up when you encounter them
- There are pros and cons for documenting the purpose of a method in Javadoc comments, please discuss with @pmbittner if this should be adopted from now on, to ease the learning curve for feature project participants and Paul himself.
